src/features/locations/handler.ts

```diff
@@ -0,0 +1,112 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import { readCSVToJSON } from '@countryconfig/features/utils'
+import { Request, ResponseToolkit } from '@hapi/hapi'
+import {
+  extractStatisticsMap,
+  getStatistics,
+  LocationStatistic
+} from '../administrative/scripts/utils'
+
+type HumdataLocation = {
+  admin0Pcode: string
+  admin0Name_en: string
+  admin0Name_alias?: string
+
+  admin1Pcode?: string
+  admin1Name_en?: string
+  admin1Name_alias?: string
+
+  admin2Pcode?: string
+  admin2Name_en?: string
+  admin2Name_alias?: string
+
+  admin3Pcode?: string
+  admin3Name_en?: string
+  admin3Name_alias?: string
+
+  admin4Pcode?: string
+  admin4Name_en?: string
+  admin4Name_alias?: string
+}
+
+type Facility = {
+  id: string
+  name: string
+  partOf: string
+  code: 'HEALTH_FACILITY' | 'CRVS_OFFICE'
+}
+
+type Location = {
+  statisticalID: string
+  name: string
+  alias: string
+  partOf: string
+  code: 'ADMIN_STRUCTURE' | 'HEALTH_FACILITY' | 'CRVS_OFFICE'
+  physicalType: 'Jurisdiction' | 'Building'
+  jurisdictionType?: typeof JURISDICTION_TYPE[number]
+  statistics?: LocationStatistic['years']
+}
+
+const JURISDICTION_TYPE = [
+  'STATE',
+  'DISTRICT',
+  'LOCATION_LEVEL_3',
+  'LOCATION_LEVEL_4',
+  'LOCATION_LEVEL_5'
+] as const
+
+export async function locationsHandler(_: Request, h: ResponseToolkit) {
+  const [humdataLocations, healthFacilities, crvsFacilities, statistics] =
+    await Promise.all([
+      readCSVToJSON<HumdataLocation[]>(
+        './src/features/locations/locations.csv'
+      ),
+      readCSVToJSON<Facility[]>(
+        './src/features/locations/health-facilities.csv'
+      ),
+      readCSVToJSON<Facility[]>('./src/features/locations/crvs-facilities.csv'),
+      getStatistics()
+    ])
+  const locations = new Map<string, Location>()
+  const statisticsMap = extractStatisticsMap(statistics)
+  humdataLocations.forEach((humdataLocation) => {
+    ;([1, 2, 3, 4] as const).forEach((locationLevel) => {
+      const id = humdataLocation[`admin${locationLevel}Pcode`]
+      if (id) {
+        locations.set(id, {
+          statisticalID: id,
+          name: humdataLocation[`admin${locationLevel}Name_en`]!,
+          alias: humdataLocation[`admin${locationLevel}Name_alias`]!,
+          partOf:
+            locationLevel == 1
+              ? 'Location/0'
+              : `Location/${humdataLocation[`admin${locationLevel - 1}Pcode`]}`,
+          code: 'ADMIN_STRUCTURE',
+          physicalType: 'Jurisdiction',
+          jurisdictionType: JURISDICTION_TYPE[locationLevel - 1],
+          statistics: statisticsMap.get(id)?.years
```

Comment:
I wonder if this would be a good time to make some of these field names a bit more descriptive 🤔 Do we need to include `code` and `physicalType` or could the receiving end add those? Some quick ideas:
- statisticalID – could it be just `id` so whomever is building this endpoint might understand to keep it always the same?
- name, alias, partOf all make sense
- I don't remember the exact purpose of code and physicalType and those probably wouldn't be self-explanatory for anyone new to this either
- juristictionType could maybe be just `type`?

Reply:
- statisticalID, makes sense to rename it. I guess it's there for legacy reasons
- jurisdictionType makes it be a bit more specific than just "type" so I'm inclined towards keeping it as it is.
- for code & physicalType, I'm hoping making the types more explicit here with
```
  code: 'ADMIN_STRUCTURE' | 'HEALTH_FACILITY' | 'CRVS_OFFICE'
  physicalType: 'Jurisdiction' | 'Building'
```
would help

Reply:
Sounds good 👍 So this endpoint would in the future return all health facilities, CRVS offices and admin structures and thus we want to have the "code" field there to differentiate between the three. Maybe in that case `code` could be named `locationType`?

Reply:
sounds good. So we are gonna be changing
```
statisticalID -> id
code -> locationType
```